File: exercism/python/bob.py
def response(hey_bob):
    hey_bob = hey_bob.strip()
    if hey_bob == "":
        return "Fine. Be that way!"
    question = hey_bob[-1] == '?'
    capitals = hey_bob.isupper()
    if question:
        if capitals:
            return "Calm down, I know what I'm doing!"
        return "Sure."
    if capitals:
        return "Whoa, chill out!"
    return "Whatever."

File: exercism/rust/bob.rs
pub fn reply(message: &str) -> &str {
    let m: &str = message.trim();
    if m.is_empty() {
        return "Fine. Be that way!";
    }
    let question: bool = m.ends_with('?');
    let capitals: bool =
        m.chars().filter(|c| c.is_alphabetic()).count() > 0 && m.to_uppercase() == m;
    if question {
        if capitals {
            return "Calm down, I know what I'm doing!";
        }
        return "Sure.";
    } else if capitals {
        return "Whoa, chill out!";
    }
    "Whatever."
}
